### PR TITLE
Setup User CRUD with PSQL DB

### PR DESCRIPTION
**Pull request by farisanadia**
[backend]
- set up controller commands for crud operations
- set up routes for api calls [frontend]
- added component for deleting, logging out, editing user
- added signup, login, user profile pages
- updated homepage to reflect new user pages

### CODE REVIEW

File: frontend/src/pages/UserProfile.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                }
+            );
+        }
+        if (loggedInUser) {
+            getUser();
+        } else {
+            //if no user session
+            navigate('/login');
+        }
+    }, [navigate, setUser]);
```
Comment (cfyjoanna) [inline frontend/src/pages/UserProfile.js:40]: check if can replace `[navigate, setUser]` with `[]`

File: frontend/src/pages/home.js [hunk 2]
```diff
@@ -14,7 +15,14 @@ function Home() {
...
     )
-  }, []);
+    //check if user is logged in
+    const loggedInUser = localStorage.getItem('user');
+    if (loggedInUser) {
+      setIsLoggedIn(true);
+    } else {
+      setIsLoggedIn(false);
+    }
+  }, [setIsLoggedIn, setBackendData]);
```
Comment (cfyjoanna) [inline frontend/src/pages/home.js:25]: check this one also (it should be the same changes as the previous one I think)

**Review by cfyjoanna — COMMENTED** [review]
looks gr8 I think. some parts are ugly but beauty is from within 💯

File: frontend/src/pages/home.js (hunk 2 above)
Comment (farisanadia) [reply frontend/src/pages/home.js]: fixed

File: frontend/src/pages/UserProfile.js (hunk 1 above)
Comment (farisanadia) [reply frontend/src/pages/UserProfile.js]: left as [navigate] per warnings given when set to []

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,18 +1,22 @@
-import './App.css';
-import React from 'react';
-import { Route, Routes } from 'react-router-dom';
-
+import Home from './pages/home';
 import Questions from './pages/questions';
-import User from './pages/user';
+import UserProfile from'./pages/UserProfile';
```
[the remark this answers is not in the document]
Comment (farisanadia) [reply frontend/src/App.js]: changed

File: backend/controllers/user-controller.js [hunk 4]
```diff
@@ -3,37 +3,133 @@ const bcrypt = require("bcrypt");
...
+
+async function deleteUser (req, res) {
+    let { email } = req.body;
+
+    pool.query(
+        `DELETE FROM users WHERE email = $1`, [email], (err, result) => {
+            if (err) {
+                console.log(err);
+            } else {
+                return res.status(200).send({message:"user deleted successfully"});
```
[the remark this answers is not in the document]
Comment (farisanadia) [reply backend/controllers/user-controller.js]: fixed

File: backend/controllers/user-controller.js [hunk 5]
```diff
@@ -3,37 +3,133 @@ const bcrypt = require("bcrypt");
...
+                bcrypt.compare(password, result.rows[0].password, (err, result) => {
+                    if (err) {
+                        throw err;
+                    }
+                    if (result) {
+                        return res.status(200).json({user});
+
+                    } else {
+                        return res.status(400).json({
+                            error:"incorrect password"
```
[the remark this answers is not in the document]
Comment (farisanadia) [reply backend/controllers/user-controller.js]: fixed

File: backend/controllers/user-controller.js [hunk 6]
```diff
@@ -3,37 +3,133 @@ const bcrypt = require("bcrypt");
...
+            if (err) {
+                throw err;
+            }
+            if (result.rows.length > 0) {
+                return res.status(400).json({
+                    error: "Email already exists."
+                });
+            } else {
+                //register user
+                pool.query(
```
[the remark this answers is not in the document]
Comment (farisanadia) [reply backend/controllers/user-controller.js]: changed

**Comment by cfyjoanna** [comment]
slayed ate devoured left no crumsb LGTM

**Pull request merged**